src/core/audio/AudioJack.cpp

```diff
@@ -204,6 +204,7 @@ void AudioJack::startProcessing()
 {
 	if( m_active || m_client == NULL )
 	{
+		m_stopped = false;
```

Comment:
This will make sense if the client is already active, but how about the null client case?

Reply:
Good question!

In modified stable code I tried to make situation with  `m_client == NULL`  true:
- if there is no jackd (package is not installed in system), then this "driver" fails to start and instead starts dummy "driver";
- if jackd available (package is installed in system), then driver start jackd itself ;
- if jackd is already started with qjackctl , then driver use started jackd; and if I stop jackd with qjackctl, then "driver" simply restart jackd   (if it was connected, than restart was correct, but if I disconnect lmms before stopping jackd, than restart ended with endless loop trying start jackd; I was able close gui, but application was closed only after terminal closing).

But anyway I can not model situation than `AudioJack::startProcessing() ` called with `m_client == NULL`  true . It seems that `m_client == NULL`  always than `AudioJack::startProcessing() `  is called. 

But from another point of view, it seems that nothing happens if `AudioJack::processCallback`  will be called  `m_client == NULL`  true . In this case (`m_client == NULL`  true) if 
`m_stopped = false;` will be replaced (for example) with `if (m_client != NULL) {m_stopped = false;}` , 
than LMMS will hung after exporting project in file.  

So, if "driver" can work with `m_client == NULL`  true - and "eat" audio data from Mixer, than code should be as it is in PR now. But if "driver" can not work in this case , than we should replace 
`if( m_active || m_client == NULL )`
with something like:

```
assert(m_client != NULL);
if( m_active )
```

I really don't know how to check this ...

Reply:
It seems, that the sentence:
"_In LMMS  `m_client == NULL` always false than `AudioJack::startProcessing()` is called._"
can be proved "mathematically". 

If this is really true, than current PR simply fix bug. 
All other changes are not needed for bug fix , but are "some code cleaning".

Reply:
On modified stable code **I have checked situation when** `m_client == NULL` **is true** (I simply change initialization code to leave object uninitialized). 

"Driver" can not work properly (when `m_client == NULL`  is true) - it is not "eating" audio data from mixer, so application hung on file export start. 

So question to maintainers: "_Should I add additional lines of code to handle situation, that can not happen ?_"
:

Reply:
I looked into the JACK source code. That situation can only happen when JACK calls `AudioJack::shutdownCallback()` on failures. In that case, `AudioJack::restartAfterZombified()` will try to restart the client. If that fails, `m_client` will be null.
Then, `AudioJack` won't consume audio anymore.

Reply:
You are right, but after this `AudioJack::startProcessing()` is not called anymore ... 
Even if we try export audio data, than LMMS will hung before real audio export process. 

P.S.

But "in real life" restart the client always not fail , except when user 
- starts jackd;
- starts LMMS;
- uninstall   jackd package (if this can be done);
- quit jackd using ( what ...., after that user have no qjackctl) command lien with su.

There for I mentioned infinite loop trying restart client. 

P.S.
So 
```
assert(m_client != NULL);
if( m_active )
```
is not the option!

Reply:
My be should be added something like:
```
if (m_client == NULL)
{
    // In this case LMMS can not work and will hung on some operation, but user still able save project.
    // If this is happened user is warned (IN AudioJack::restartAfterZombified).
    // TODO : when will be possible change driver "on the fly" this code should be rewritten ! 
    fprintf(stderr, "AudioJack::startProcessing : ERROR : m_client == NULL : driver can not be started !!!\n");
}
```

Reply:
If there is no change request from maintainers, than this PR version is **final** - ready to merge. 

P.S.

`AudioJack.cpp`  code should be improved but not in bug fix context. (for example, in "When Jack is used, start / stop jack transport" context)

Reply:
It seems like `m_stopped` doesn't change anything if the client is null. I think it's safe to merge.